File: leet_BS_658_find_k_closest_elements___twopointer.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x)
    {
        vector<int> res;
        int n = arr.size();
        int right = n - 1;
        int left = 0;
        while (left < right && n > k)
        {
            if (abs(arr[left] - x) > abs(arr[right] - x))
                ++left;
            else
                --right;
            --n;
        }
        //cout << left << ' ' << right << endl;
        while (left <= right){
            res.push_back(arr[left]);
            left++;
        }
        return res ;
    }
};

File: leet_BS_658_find_k_closest_elements___twopointer.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        n = len(arr)
        right = n - 1
        left = 0
        while left < right and n > k:
            if abs(arr[left] - x) > abs(arr[right] - x):
                left += 1
            else:
                right -= 1
            n -= 1
        res = []
        while left <= right:
            res.append(arr[left])
            left += 1
        return res
